sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -3039,3 +3039,113 @@ case class SplitPart (
+    numbers to distinguish valid numbers from mistyped, incorrect numbers.
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_('8112189876');
+       true
+      > SELECT _FUNC_('79927398713');
+       true
+      > SELECT _FUNC_('79927398714');
+       false
+  """,
+  since = "3.4.0",
+  group = "string_funcs")
+case class Luhncheck(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+
+  override def nullable: Boolean = false
+
+  override protected def withNewChildInternal(newChild: Expression): Luhncheck =
+    copy(child = newChild)
+
+  /**
+   * Expected input types from child expressions. The i-th position in the returned seq indicates
+   * the type requirement for the i-th child.
+   *
+   * The possible values at each position are:
+   *   1. a specific data type, for example, LongType, StringType.
+   *   2. a non-leaf abstract data type, for example, NumericType, IntegralType, FractionalType.
+   */
+  override def inputTypes: Seq[AbstractDataType] = Seq(StringType)
+
+  /**
+   * Returns the [[DataType]] of the result of evaluating this expression. It is invalid to query
+   * the dataType of an unresolved expression (i.e., when `resolved` == false).
+   */
+  override def dataType: DataType = BooleanType
+
+  /**
+   * Default behavior of evaluation according to the default nullability of UnaryExpression. If
+   * subclass of UnaryExpression override nullable, probably should also override this.
+   */
+  override def eval(input: InternalRow): Any = Luhncheck.isLuhnNumber(child.eval(input))
+
+  /**
+   * Returns Java source code that can be compiled to evaluate this expression. The default
+   * behavior is to call the eval method of the expression. Concrete expression implementations
+   * should override this to do actual code generation.
+   *
+   * @param ctx
+   *   a [[CodegenContext]]
+   * @param ev
+   *   an [[ExprCode]] with unique terms.
+   * @return
+   *   an [[ExprCode]] containing the Java source code to generate the given expression
+   */
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    defineCodeGen(
+      ctx,
+      ev,
+      c => {
+        s"""org.apache.spark.sql.catalyst.expressions.Luhncheck.isLuhnNumber($c)"""
+      })
+  }
+}
+
+object Luhncheck {
+
+  /**
+   * Function to check if a given number string is a valid Luhn number
+   *
+   * @param numberString
+   *   the number string to check
+   * @return
+   *   true if the number string is a valid Luhn number, false otherwise
+   */
+  def isLuhnNumber(numberString: Any): Boolean = {
+    if (numberString != null) {
+      val digits = numberString.asInstanceOf[UTF8String].toString
+      // check if all characters in the input string are digits
+      if (!digits.forall(_.isDigit)) {
+        return false // if not, return false
+      }
+      // reverse the string so that we can use a foldLeft function
```

Comment:
please update the comment to start with a capital letter and end with punctuation (here and everywhere else in the PR)

Reply:
Done